### PR TITLE
fix(mac): display package info after keyboard installation

### LINKED ISSUE #9308: bug(mac): Package Information dialog not displayed after keyboard installation with macOS Ventura
### Describe the bug

I noticed that the Package Information dialog is missing during the Keyboard installation using Keyman Configuration dialog. 

### Reproduce the bug

1. Install Keyman 17.0.142-alpha build. 
2. Open Keyman Configuration dialog. 
3. Click  Download Keyboard button. 
4. Install IPA (SIL) or any keyboard. 
We may notice that Download Complete dialog appears with Done button. 
5. Click the Done button. 

Here, I noticed that after clicking the Done button, I did not see any Installation dialog on the screen. Again, I opened the Configuration dialog to see the 'package information' page. Seems to be an issue. 

Note: Tried the same scenario in Mojave macOS (VM) and BigSur macOS (VM) and I was not able to reproduce it. 

I have attached the Video file for reference.                                                     

https://github.com/keymanapp/keyman/assets/19683143/10db80d0-d245-4f46-be02-0f3a8e9ab08c


                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        

### Expected behavior

The Package Information page should appears after clicking the Done button from Download Complete dialog. 

### Related issues

_No response_

### Keyman apps

- [ ] Keyman for Android
- [ ] Keyman for iPhone and iPad
- [ ] Keyman for Linux
- [ ] Keyman for macOS
- [X] Keyman for Windows
- [ ] Keyman Developer
- [ ] KeymanWeb
- [ ] Other - give details at bottom of form

### Keyman version

17.0.142-alpha

### Operating system

macOS 13.4 Ventura

### Device

_No response_

### Target application

_No response_

### Browser

_No response_

### Keyboard name

_No response_

### Keyboard version

_No response_

### Language name

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by sgschantz**
When a keyboard is downloaded and installed, the package information window should be displayed immediately after dismissing the download dialog:

image

Fixes #9308

# User Testing

## TEST_PACKAGE_INFO_DISPLAYED

1. Open Keyman Configuration dialog. 
1. Click Download Keyboard button. 
1. Install any keyboard. 
1. Confirm that the Package Information dialog is displayed after clicking the Done button. 

## TEST_DELETE_KEYBOARD

1. Open Keyman Configuration dialog. 
1. For an installed keyboard, click on the delete icon  
1. Confirm that a confirmation dialog is correctly displayed
1. Click the Delete button on the dialog 
1. Confirm that keyboard has been deleted and is no longer listed in the configuration window or in the Keyman menu 

## TEST_BOGUS_KEYBOARD_INSTALL

1. In the Finder, rename a small text file to bogus.kmp
1. Double-click the file 
1. Confirm that an installation confirmation dialog is correctly displayed
1. Click the Install button
1. Confirm that an error alert is displayed, indicating that the .kmp file could not be read

### CODE REVIEW

File: mac/Keyman4MacIM/Keyman4MacIM/KMConfiguration/KMConfigurationWindowController.m [hunk 1]
```diff
@@ -409,19 +409,22 @@ - (void)helpAction:(id)sender {
...
-                                   modalDelegate:self
-                                  didEndSelector:@selector(alertDidEnd:returnCode:contextInfo:)
-                                     contextInfo:(__bridge void *)([NSNumber numberWithInteger:removeButton.tag])];
+  [self.deleteAlertView beginSheetModalForWindow:self.window completionHandler:^(NSModalResponse returnCode)  {
+    if (returnCode == NSAlertFirstButtonReturn) {
+      os_log_debug([KMLogs uiLog], "confirm delete keyboard alert dismissed");
+      [self deleteFileAtIndex:[NSNumber numberWithInteger:deleteButton.tag]];
+      self.deleteAlertView = nil;
+    }
+  }];
```
Comment (mcdurdin) [inline mac/Keyman4MacIM/Keyman4MacIM/KMConfiguration/KMConfigurationWindowController.m:427]: Is this unrelated?

File: mac/Keyman4MacIM/Keyman4MacIM/KMConfiguration/KMConfigurationWindowController.m [hunk 2]
```diff
@@ -449,25 +452,29 @@ - (IBAction)useVerboseLoggingCheckBoxAction:(id)sender {
...
   
-  [self.confirmKmpInstallAlertView beginSheetModalForWindow:self.window
-                                              modalDelegate:self
-                                             didEndSelector:@selector(alertDidEnd:returnCode:contextInfo:)
-                                                contextInfo:(__bridge void *)(package)];
+  [self.confirmKmpInstallAlertView beginSheetModalForWindow:self.window completionHandler:^(NSModalResponse returnCode)  {
+    os_log_debug([KMLogs uiLog], "confirm keyboard installation alert dismissed");
+    if (returnCode == NSAlertFirstButtonReturn) {
+      [self installPackageFile: package.getTempKmpFilename];
+      self.confirmKmpInstallAlertView = nil;
```
Comment (mcdurdin) [inline mac/Keyman4MacIM/Keyman4MacIM/KMConfiguration/KMConfigurationWindowController.m:465]: Does this mean that `self.confirmKmpInstallAlertView` is not nilled if another returnCode is given?

File: mac/Keyman4MacIM/Keyman4MacIM/KMConfiguration/KMConfigurationWindowController.m [hunk 3]
```diff
@@ -558,6 +564,7 @@ - (NSAlert *)confirmKmpInstallAlertView {
   return _confirmKmpInstallAlertView;
 }
 
+/*
```
Comment (mcdurdin) [inline mac/Keyman4MacIM/Keyman4MacIM/KMConfiguration/KMConfigurationWindowController.m:567]: Chore: delete commented code

**Review by mcdurdin — APPROVED** [review]

File: mac/Keyman4MacIM/Keyman4MacIM/KMConfiguration/KMConfigurationWindowController.m (hunk 1 above)
Comment (sgschantz) [reply mac/Keyman4MacIM/Keyman4MacIM/KMConfiguration/KMConfigurationWindowController.m:427]: Not related to keyboard installation, but for this fix I replaced all the deprecated calls to beginSheetModalForWindow with the newer version that behaves slightly differently. The older API was deprecated in macOS 10.10 and we now require 10.13.

File: mac/Keyman4MacIM/Keyman4MacIM/KMConfiguration/KMConfigurationWindowController.m (hunk 2 above)
Comment (sgschantz) [reply mac/Keyman4MacIM/Keyman4MacIM/KMConfiguration/KMConfigurationWindowController.m]: Yes, but it should be. I didn't move that over correctly when I created the new handler. I've changed it now.

**Review by mcdurdin — APPROVED** [review]

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 18.0.104-alpha](https://keyman.com/downloads/releases/alpha/18.0.104)

**Pull request merged**